toolz/itertoolz.py

```diff
@@ -495,6 +496,21 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def cycles(n, seq):
+    """ Cycles through the sequence n-times.
+
+    Note: If ``n`` is ``None``, cycle infinitely.
+
+    >>> list(cycles(2, [1, 2, 3]))
+    [1, 2, 3, 1, 2, 3]
+    """
+
+    if n is None:
```

Comment:
What is the motivation for this special case, if a user was to get None, shouldn't they just call `itertools.cycle`?